semeio/workflows/ahm_analysis/ahmanalysis.py

```diff
@@ -0,0 +1,466 @@
+
+
+def create_path(runpathdf, target_name):
+    """to create path for output data"""
+    return tempfile.mkdtemp()
+
+
+def calc_observationsgroup_misfit(obs_keys, df_update_log, misfit_df):
+    """To get the misfit for total observations (active/inactive)."""
+    misfit_values = []
+    total_obs_nr = len(df_update_log[df_update_log.status.isin(["Active", "Inactive"])])
+    for key in obs_keys:
+        misfit_name = "MISFIT:" + key
+        if misfit_name in misfit_df:
+            misfit_values.append(misfit_df[misfit_name].mean())
+        else:
+            warnings.warn("WARNING: no MISFIT value for observation " + key)
+            misfit_values.append(0)
+    return np.sum(misfit_values) / total_obs_nr
+
+
+def load_grid_to_dataframe(grid_path):
+    """Get field grid characteristics/coordinates"""
+    grid_path = Path(grid_path).with_suffix("")
+    try:
+        grid = Grid(grid_path, fformat="eclipserun")
+        return grid.dataframe(activeonly=False)
+    except OSError as err:
+        raise OSError("A grid with .EGRID format is expected.") from err
+
+
+def calc_mean_delta_grid(all_input_post, all_input_prior, grid):
+    """calculate mean delta of field grid data"""
+    delta_post_prior = np.subtract(all_input_prior, all_input_post)
+    delta_post_prior = np.absolute(delta_post_prior)
+    # Can we avoid changing in place here?
+    grid["Mean_delta"] = np.mean(delta_post_prior, axis=0)
+    df_mean_delta = grid.groupby(["IX", "JY"])[["Mean_delta"]].mean().reset_index()
+
+    return df_mean_delta["Mean_delta"]
+
+
+def _filter_on_prefix(list_of_strings, prefixes):
+    """returns the set of strings that has a match for any of the given prefixes"""
+    return {
+        string
+        for string in list_of_strings
+        if any(string.startswith(prefix) for prefix in prefixes)
+    }
+
+
+def get_updated_parameters(prior_data, parameters):
+    """make list of updated parameters
+    (excluding duplicate transformed parameters)
+    """
+    parameter_keys = _filter_on_prefix(
+        list_of_strings=prior_data.keys(), prefixes=parameters
+    )
+    # remove parameters with constant prior distribution
+    p_keysf = []
+    for dkey in parameter_keys:
+        if prior_data[dkey].ndim > 1:
+            warnings.warn("WARNING: Parameter " + dkey + " defined several times.")
+            flatten_arr = np.ravel(prior_data[dkey])
+            result = np.all(prior_data[dkey] == flatten_arr[0])
+            if not result:
+                p_keysf.append(dkey)
+        else:
+            if not all(x == prior_data[dkey][0] for x in prior_data[dkey]):
+                p_keysf.append(dkey)
+    return p_keysf
+
+
+def calc_kolmogorov_smirnov(columns, prior_data, target_data):
+    """Calculate kolmogorov_smirnov matrix"""
+    ks_param = {}
+    for dkey in sorted(columns):
+        ks_val = ks_2samp(prior_data[dkey], target_data[dkey])
+        ks_param[dkey] = ks_val[0]
+    #        ks_matrix.at[dkey, key] = ks_val[0]
```

Comment:
Suppose the comment here can be removed, guess it snuck inn during refactor :)

Reply:
yes :-)